std/format/internal/write.d

```diff
@@ -2546,9 +2546,23 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
             {
                 // ignore hidden context pointer
             }
-            else static if (0 < i && T.tupleof[i-1].offsetof == T.tupleof[i].offsetof)
+            /* https://github.com/dlang/phobos/issues/10840
+             * handle possible bitfields by doing overlap comparisons
+             * using bit counts rather than byte counts.
+             * However, the overlap
+             * check in general does not take into account staggered unions.
+             * This can be fixed using the correct algorithm implemented in
+             * the compiler function dmd.declaration.isOverlappedWith().
```

Comment:
FYI tracking this issue with staggered unions in https://github.com/dlang/phobos/issues/10844

Reply:
good!